### DISCUSSION TITLE
brew services start, launchctl bootstrap, Input/output error, php-fpm

### DISCUSSION THREAD

**Discussion by craigfrancis** [Homebrew/discussions#1372] [Everyday usage]
Might a new general issue with launchctl in MacOS 11.3 (20E232).

When running `brew services start php` it complains with:

    Bootstrap failed: 5: Input/output error
    Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/x/Library/LaunchAgents/homebrew.mxcl.php.plist` exited with 5.

If you watch `/var/log/system.log` this isn't exactly an Input/output error, the error message is about ProgramArguments:

    entering bootstrap mode
    homebrew.mxcl.php (lint): Service specifies neither Program nor ProgramArguments.
    Could not import service from caller: path = /Users/x/Library/LaunchAgents/homebrew.mxcl.php.plist, caller = launchctl.4878, error = 111: Invalid or missing Program/ProgramArguments
    exiting bootstrap mode

The current "homebrew.mxcl.php.plist" file contains:

    <key>ProgramArguments</key>
    <string>/usr/local/opt/php/sbin/php-fpm</string>

Changing this into an `<array>`, e.g.

    <key>ProgramArguments</key>
    <array>
        <string>/usr/local/opt/php/sbin/php-fpm</string>
    </array>

Seems to fix the issue, where you then call the `/bin/launchctl bootstrap ...` command yourself.

But the plist file is reset if using `brew services start php`.

**Comment by SMillerDev** [comment]
It seems a recent fix of mine broke this. I'll look into it.

**Comment by robertmcintyre22** [comment]
Hi, 
I'm getting a the same issue when trying to run `brew services start php` - has a fixed be found and released?

**Reply by craigfrancis** [reply]
I don't think so, but if you edit `~/Library/LaunchAgents/homebrew.mxcl.php.plist` so the line after `<ProgramArguments>` has been changed as per the example above (adding in `<array>` and  `</array>`), then use the `/bin/launchctl bootstrap ...` command (because `brew services start` will reset your changes), then it should give you a temporary work around until it is fixed.

**Reply by SMillerDev** [reply]
https://github.com/Homebrew/brew/pull/11279 fixes it, but that hasn't been fully release yet. You could do `git -C $(brew --repo) pull origin master` to get the pre-release commits from the branch.

**Reply by craigfrancis** [reply]
Thanks @SMillerDev, really appreciate it, along with everything Homebrew does (it makes my life so much easier).

**Reply by robertmcintyre22** [reply]
Thanks @craigfrancis and @SMillerDev for the quick helpful replies and suggestions, much appreciated.
I will give the temporary fix ago until the fix is released. Thanks again.

**Reply by SMillerDev** [reply]
Fix has just been released.

**Reply by craigfrancis** [reply]
Thanks, can confirm this works :-)

    <key>ProgramArguments</key>
    <array>
        <string>/usr/local/opt/php/sbin/php-fpm</string>
        <string>--nodaemonize</string>
    </array>

**Comment by mylonelycomputer** [comment]
Hello, I've pulled the changes and I have the above reflected in the `plist` file but when running `brew start services php@7.2` I still get the same:

```Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/502 ~/Library/LaunchAgents/homebrew.mxcl.php@7.2.plist` exited with 5.
